src/d3a/models/market/two_sided.py

```diff
@@ -19,21 +19,33 @@
 from dataclasses import replace
 from logging import getLogger
 from math import isclose
-from typing import Union  # noqa
+from typing import Union, List, Dict  # noqa
 
+from d3a.constants import FLOATING_POINT_TOLERANCE
+from d3a.models.market.market_validators import RequirementsSatisfiedChecker
 from d3a_interface.constants_limits import ConstSettings
 
-from d3a.d3a_core.exceptions import BidNotFound, InvalidBid, InvalidTrade, MarketException
+from d3a.d3a_core.exceptions import (
+    BidNotFoundException, InvalidBid, InvalidTrade, MarketException,
+    InvalidBidOfferPair)
 from d3a.d3a_core.util import short_offer_bid_log_str
 from d3a.events.event_structures import MarketEvent
-from d3a.models.market import lock_market_action, validate_authentic_bid_offer_pair
-from d3a.models.market.market_structures import Bid, Trade, TradeBidOfferInfo
+from d3a.models.market import lock_market_action
+from d3a.models.market.market_structures import Bid, Trade, TradeBidOfferInfo, Offer
 from d3a.models.market.one_sided import OneSidedMarket
 
 log = getLogger(__name__)
 
 
 class TwoSidedMarket(OneSidedMarket):
+    """Extends One sided market class and adds support for bidding functionality.
+
+    A market type that allows producers to place energy offers to the markets
+    (exactly the same way as on the one-sided market case), but also allows the consumers
+    to place energy bids on their respective markets.
+    Contrary to the one sided market, where the offers are selected directly by the consumers,
+     the offers and bids are matched by the myco matcher.
```

Comment:
```suggestion
     the offers and bids are being matched via some matching algo.
```
Myco is an extension of two sided market, there are already 2 traditional TwoSidedMarket based matching algos (pay-as-bid/pay-as-clear)